src/vma/util/utils.cpp

```diff
@@ -792,6 +795,90 @@ size_t get_local_ll_addr(IN const char * ifname, OUT unsigned char* addr, IN int
 
 	return bytes_len; // success
 }
+bool check_bond_device_exist(const char* ifname)
+{
+#ifdef HAVE_LIBNL3
+	int ret = 0;
+	struct nl_cache *cache = NULL;
+	struct rtnl_link *link = NULL;
+	char *link_type = NULL;
+
+	struct nl_sock *nl_socket = nl_socket_alloc();
+	if (!nl_socket) {
+		goto out;
+	}
+	nl_socket_set_local_port(nl_socket, 0);
+	ret = nl_connect(nl_socket, NETLINK_ROUTE);
+	if (ret < 0) {
+		goto out;
+	}
+	ret = rtnl_link_alloc_cache(nl_socket, AF_UNSPEC, &cache);
+	if (!cache) {
+		goto out;
+	}
+	link = rtnl_link_get_by_name(cache, ifname);
+	if (!link) {
+		goto out;
+	}
+	link_type = rtnl_link_get_type(link);
+	if (link_type &&
+			strcmp(link_type, "bond") && strcmp(link_type, "team")) {
+		link_type = NULL;
+	}
+out:
+	if (cache) {
+		nl_cache_free(cache);
+	}
+	if (nl_socket) {
+		nl_socket_free(nl_socket);
+	}
+
+	return ((bool)link_type);
+#else
+	char device_path[256] = {0};
+	sprintf(device_path, BOND_DEVICE_FILE, ifname);
+	int fd = orig_os_api.open(device_path, O_RDONLY);
+	if (fd >= 0)
+		orig_os_api.close(fd);
+	if (fd < 0 && errno == EMFILE) {
+		__log_warn("There are no free fds in the system. This may cause unexpected behavior");
+	}
+
+	return (fd > 0);
```

Comment:
Please, replace this code with a call to `check_device_exist()`. Also replace `sprintf()` with secure `snprintf()`.